### PR TITLE
feat(custom): reusable custom series

### PR DESCRIPTION
**Pull request by Ovilia**
## Brief Information

This pull request is in the type of:

- [ ] bug fixing
- [x] new feature
- [ ] others

### What does this PR do?

Custom series provide a powerful way to let developers define their own charts. But for now, it's hard to reuse the `renderItem` for other projects or other developers. This PR provides a way to register custom series so that developer A can make a custom series and publish it (probably by npm) and developer B can use it by simply setting `renderItem` to the related name.

### Before: What was the problem?

Custom series is hard to be reused.

#### Making a reusable custom series:

```ts
const renderItem = (params, api) => {
    return {
        type: 'circle',
        shape: {
            cx: api.coord([api.value(0), api.value(1)])[0],
            cy: api.coord([api.value(0), api.value(1)])[1],
            r: api.value(2) * (params.itemPayload.scale || 1)
        },
        style: {
            fill: api.visual('color'),
            opacity: params.itemPayload.opacity() || 1,
        }
    }
};
echarts.registerCustomSeries('bubble', renderItem);
```

#### Using the above custom series:

```ts

const option = {
    xAxis: {},
    yAxis: {},
    series: {
        type: 'custom',
        renderItem: 'bubble',
        itemPayload: {
            scale: 2,
            opacity: () => Math.random() * 0.5 + 0.5
        },
        data: [[11, 22, 20], [33, 44, 40], [18, 24, 10]]
    }
};
```

## Document Info

One of the following should be checked.

- [ ] This PR doesn't relate to document changes
- [x] The document should be updated later
- [ ] The document changes have been made in apache/echarts-doc#xxx

### ZRender Changes

- [ ] This PR depends on ZRender changes (ecomfe/zrender#xxx).

### Related test cases or examples to use the new APIs

N.A.

### Merging options

- [ ] Please squash the commits into a single one when merging.

### CODE REVIEW

File: src/model/Global.ts [hunk 1]
```diff
@@ -578,6 +580,10 @@ echarts.use([${seriesImportName}]);`);
         }
     }
 
+    getCustomRenderer(type: string) {
+        return getCustomSeries(type);
+    }
+
```
Comment (100pah) [inline src/model/Global.ts:586]: `ecModel.getCustomRenderer` doesn't seem to be necessary.

`CustomView` can access `customSeriesRegister` and call `getCustomRenderer` directly.

**Review by 100pah — COMMENTED** [review]
Do we need to enable it to pass the additional user params to `renderItem`, as discussed earlier?

Custom rendering logic provider:
```js
function myRenderItem(params) {
    console.log(  params.renderItemArgs );
    // params.renderItemArgs is used in the following rendering logic.
}
echarts.registerCustomSeries('perfectRenderItem', myRenderItem);
```

Custom rendering logic user:
```js
option = {
  series: {
    type: 'custom', 
    renderItem: 'perfectRenderItem'
    renderItemArgs: {abc: 123, def: 'haha'}
  }
}
```

**Review by 100pah — APPROVED** [review]
LGTM

File: src/chart/custom/customSeriesRegister.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { CustomSeriesRenderItem } from './CustomSeries';
```
Comment (pissang) [inline src/chart/custom/customSeriesRegister.ts:1]: Use `import type` explictly to ensure the `CustomSeries` is not in the depedencies of `core/echarts`
```ts
import type { CustomSeriesRenderItem } ...
```

**Review by 100pah — APPROVED** [review]

**Review by pissang — APPROVED** [review]

**Pull request merged**